auto_run.py

```diff
@@ -0,0 +1,146 @@
+#%% 
+import hourglass
+import imp
+imp.reload(hourglass)
+from hourglass import HourglassNet
+from constants import *
+import matplotlib.pyplot as plt
+import os
+
+import data_generator
+imp.reload(data_generator)
+import data_generator
+import time
+
+import re
+import pandas as pd
+import evaluation
+from evaluation import Evaluation
+imp.reload(evaluation)
+from HeatMap import HeatMap
+
+#import HeatMap
+#imp.reload(HeatMap)
+
+import numpy as np
+
+#%% find duplicates in list
+
+def find_dup(pair_list, element):
+    for (a,b) in pair_list:
+        if element == a or element == b:
+            return True
+    return False
+
+#%% 
+
+def find_epochs(base_dir, sub_dir, epoch_dic, visited_models):
+    directory = os.listdir(os.path.join(base_dir, sub_dir))
+    weight_file = [f for f in directory if (f.endswith(".hdf5"))]
+    for name in weight_file:
+        match = re.search('hpe_epoch(\d+)', name)
+        if match:
+            epoch_dic.append((match.group(1), sub_dir))
+
+    resume_file = [f for f in os.listdir(base_dir) if (sub_dir in f and f is not sub_dir)]
+    for file in resume_file:
+        visited_models.append(file)
+        for name in os.listdir(os.path.join(base_dir, file)):
+            match = re.search('hpe_epoch(\d+)', name)
+            if match and not find_dup(epoch_dic, match.group(1)):
+                epoch_dic.append((match.group(1), file))
+
+    return list(set(epoch_dic))
+
+
+#%% save stacked evaluation heatmaps
+
+def stacked_eval_heatmaps(test_df,eval):
+
+    generator = data_generator.DataGenerator(
+            df=test_df,
+            base_dir=DEFAULT_VAL_IMG_PATH,
+            input_dim=INPUT_DIM,
+            output_dim=OUTPUT_DIM,
+            num_hg_blocks=eval.num_hg_blocks,
+            shuffle=False,  
+            batch_size=len(test_df),
```

Comment:
If we only ever plan on passing the representative set, then this fine, but if we pass a `test_df` of the scale similar to our test set then it won't fit in memory. Will have to visualize in batches.

Reply:
I think we should just past in the representative set for quick comparisons. It takes about 40 mins to run 10 epochs on the representative set for each model.